covid_moonshot/analysis.py

```diff
@@ -27,31 +28,82 @@ def mask_outliers(a: np.array, max_value: float, n_devs: float):
     return (np.abs(a) < max_value) & (np.abs(a - np.mean(a)) < n_devs * np.std(a))
 
 
-def get_phase_analysis(
-    works: List[Work], max_work_value=1e4, max_n_devs=5, min_num_work_values=10
-):
+def filter_work_values(
+    f_works: np.array,
+    r_works: np.array,
+    max_work_value: float = 1e4,
+    max_n_devs: float = 5,
+) -> Tuple[np.array, np.array, int]:
+    """Remove pairs of works when either is determined to be an outlier.
 
-    f_works = np.array([w.forward_work for w in works])
-    r_works = np.array([w.reverse_work for w in works])
+    Parameters
+    ----------
+    f_works: array
+        forward works
+    r_works: array
+        reverse works
 
-    # remove pairs of works when either is determined to be an outlier
+    Returns
+    -------
+    tuple
+        tuple of forward works, reverse_works, and number of remaining work values
+    """
     f_good = mask_outliers(f_works, max_value=max_work_value, n_devs=max_n_devs)
     r_good = mask_outliers(r_works, max_value=max_work_value, n_devs=max_n_devs)
     both_good = f_good & r_good
-    f_works_good = f_works[both_good]
-    r_works_good = r_works[both_good]
     num_work_values = both_good.astype(int).sum().item()
+    return f_works[both_good], r_works[both_good], num_work_values
+
+
+def get_bar_overlap(f_works: np.array, r_works: np.array) -> float:
+    """
+    Compute the overlap (should be in [0, 1] where close to 1 is good, close to 0 bad).
+
+    Parameters
+    ----------
+    f_works: array
+        forward works
+    r_works: array
+        reverse works
+
+    Returns
+    -------
+    float
+        overlap
+    """
+
+    u_kn = np.block(
```

Comment:
this is treated the same as a numpy 2xN array right?

Reply:
Yeah, but I agree that this is not very clear. I refactored a bit (6fb6635) to just pass around a 2xN array, because as it is we are constraining the length of the forward and reverse work arrays to be the same.